File: frontend/src/types/workflow.ts
import type { Node, Edge } from '@xyflow/react';

export interface Workflow {
  id: number;
  name: string;
  description?: string;
  graphJson: {
    nodes: Node[];
    edges: Edge[];
  };
  createdAt: string;
  updatedAt: string;
}

export interface WorkflowListItem {
  id: number;
  name: string;
  description?: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/src/main/java/com/paiagent/workflow/entity/Workflow.java
package com.paiagent.workflow.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.baomidou.mybatisplus.extension.handlers.JacksonTypeHandler;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.Map;

@Data
@TableName(value = "pai_workflow", autoResultMap = true)
public class Workflow {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private String name;

    private String description;

    @TableField(typeHandler = JacksonTypeHandler.class)
    private Map<String, Object> graphJson;

    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
